### PR TITLE
Add tabs to custom dashboard listing page

### PR DESCRIPTION
**Pull request by akashmane1598**
## Description
Added tabs to the custom dashboard listing page.
Screenshot 2022-08-10 at 4 41 31 PM

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/observability/src/pages/custom-dashboards/custom-dashboards.component.ts [hunk 1]
```diff
@@ -88,14 +81,21 @@ export class CustomDashboardListComponent {
...
-
+  public dashboardView: DashboardViewType = DASHBOARD_VIEWS.MY_DASHBOARDS;
   public constructor(
     private readonly customDashboardService: CustomDashboardService,
     private readonly navigationService: NavigationService,
     private readonly activatedRoute: ActivatedRoute,
     private readonly changeDetectorRef: ChangeDetectorRef,
     private readonly userInfoService: UserInfoService
   ) {
+    this.activatedRoute.params.subscribe(params => {
```
Comment (cseas) [inline projects/observability/src/pages/custom-dashboards/custom-dashboards.component.ts]: Add subscriptionLifecycle

**Review by cseas — COMMENTED** [review]
Please deploy changes to stage for testing.

**Comment by akashmane1598** [comment]
> Please deploy changes to stage for testing.

Waiting for checks to pass  :sweat_smile:

**Review by cseas — APPROVED** [review]
LGTM. The [save/cancel buttons][1] can be handled in another PR. Please update API fields to camelCase before merging.

[1]: https://razorpay.slack.com/archives/C03A6MZF5U4/p1660131166205229

File: projects/observability/src/pages/custom-dashboards/custom-dashboard-detail/custom-dashboard-detail.component.ts [hunk 2]
```diff
@@ -68,6 +75,7 @@ export class CustomDashboardDetailComponent {
     this.activedRoute.params.subscribe(params => {
       this.dashboardId = params.dashboard_id;
       this.isNew = params.dashboard_id === 'create';
+      this.dashboardView = params.dashboard_view;
```
Comment (jaywalker21) [inline projects/observability/src/pages/custom-dashboards/custom-dashboard-detail/custom-dashboard-detail.component.ts:96]: Do you rather want to save a derived variable here? Something like below, given we are doing the `if` check 2-3 times in the component.

```javascript
this.isMyDashboardTab = params.dashboard_view === DASHBOARD_VIEWS.MY_DASHBOARDS;
```

That can help remove `if` condition from everywhere in the page.

File: projects/observability/src/pages/custom-dashboards/custom-dashboards-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      label: 'My Dashboards',
+      value: DASHBOARD_VIEWS.MY_DASHBOARDS
+    },
+    {
+      label: 'All Dashboards',
+      value: DASHBOARD_VIEWS.ALL_DASHBOARDS
+    }
+  ];
+  public constructor(private readonly navigationService: NavigationService) {
+    this.currentContext = this.contextItems[0];
```
Comment (jaywalker21) [inline projects/observability/src/pages/custom-dashboards/custom-dashboards-view.component.ts]: I think we should read from `url` here.
If I am at `https://hypertrace.concierge.stage.razorpay.in/custom-dashboards/all-dashboards?time=1h` and I reload the page, it always redirects or resets to `my-dashboards`.

What are your thoughts?

**Review by jaywalker21 — COMMENTED** [review]
Few comments @akashmane1598. Not much huge.

**Review by jaywalker21 — APPROVED** [review]

**Pull request merged**